src/Site.js

```diff
@@ -813,6 +814,38 @@ Site.prototype.deploy = function () {
         options.branch = this.siteConfig.deploy.branch || defaultDeployConfig.branch;
         options.message = this.siteConfig.deploy.message || defaultDeployConfig.message;
         options.repo = this.siteConfig.deploy.repo || defaultDeployConfig.repo;
+
+        if (travisTokenVar) {
+          if (!process.env.TRAVIS) {
+            reject(new Error('-t/--travis should only be run in Travis CI.'));
+            return undefined;
+          }
+          // eslint-disable-next-line no-param-reassign
+          travisTokenVar = _.isBoolean(travisTokenVar) ? 'GITHUB_TOKEN' : travisTokenVar;
+          if (!process.env[travisTokenVar]) {
+            reject(new Error(`The environment variable ${travisTokenVar} does not exist.`));
+            return undefined;
+          }
+
+          const githubToken = process.env[travisTokenVar];
+          let repoSlug = process.env.TRAVIS_REPO_SLUG;
+          if (options.repo) {
+            // Extract repo slug from user-specified repo URL so that we can include the access token
+            const repoSlugRegex = /github\.com[:/]([\w-]+\/[\w-]+)\.git/;
+            const repoSlugMatch = repoSlugRegex.exec(options.repo);
+            if (!repoSlugMatch) {
+              reject(new Error(`The specified repository ${options.repo} is not a valid GitHub repository.`));
```

Comment:
Our normal deploy works for different git providers. Could we make this error more specific for the travis build? Something like:

`Deploy with travis expects a GitHub repository. The specified repository ${options.repo} is invalid.`